### PR TITLE
Refactor from_nodes into smaller functions

### PR DESCRIPTION
**Pull request by marcenacp**

### CODE REVIEW

File: python/format/src/computations.py [hunk 1]
```diff
@@ -184,8 +207,14 @@ def __call__(self):
 class ReadField(Operation):
     """Reads a field from a Pandas DataFrame and applies transformations."""
 
+    node: Field
+
     def __call__(self, args: pd.Series):
-        _, field = self.node.source.reference
+        assert len(self.node.source.reference) == 2, (
```
[the remark this answers is not in the document]
Comment (marcenacp) [reply python/format/src/computations.py:213]: Thanks for the feedback, Pieter!

There isn't a clear decision if we should fail fast (raise errors) or gather as many issues as possible before failing. So I created: https://github.com/mlcommons/datasets_format/issues/39

File: python/format/src/computations.py [hunk 2]
```diff
@@ -197,6 +226,72 @@ def __call__(self, args):
         return {self.node.name: args}
 
 
+def _find_record_set(graph: nx.MultiDiGraph, node: Node) -> RecordSet:
+    parent_node = graph.nodes[node].get("parent")
```
Comment (pierrot0) [inline python/format/src/computations.py:230]: 1line docstrings on each function may help readability.

**Review by pierrot0 — APPROVED** [review]
Thank you!

File: python/format/src/computations.py (hunk 2 above)
Comment (marcenacp) [reply python/format/src/computations.py:362]: Done.

Comment (marcenacp) [reply python/format/src/computations.py:362]: (See next PR)

**Pull request merged**